server/src/api/routes/private/private.ts

```diff
@@ -143,6 +143,29 @@ export const PrivateRouter = new Hono<Context>()
 
         await db.insert(matchDataTable).values(matchData);
         await logPlayerIPs(matchData);
+        const playerIds = matchData.map(d => d.userId).filter(d => d != null);
+        const playerWins = await db
+
+            .select({
+                userId: matchDataTable.userId,
+                wins: sql<number>`SUM(CASE WHEN ${matchDataTable.rank} = 1 THEN 1 ELSE 0 END)`.as("wins"),
+            })
+            .from(matchDataTable)
+            .where(inArray(matchDataTable.userId, playerIds))
+            .groupBy(matchDataTable.userId);
+
+        const items = playerWins.filter(({wins}) => wins >= 500)
```

Comment:
This can be written as part of the database query for a non-trivial improvement in performance.